transportable-udfs-avro/src/main/java/com/linkedin/transport/avro/StdUdfWrapper.java

```diff
@@ -68,12 +68,27 @@ protected boolean containsNullValuedNonNullableArgument(Object[] arguments) {
     return false;
   }
 
-  protected StdData wrap(Object avroObject, StdData stdData) {
-    if (avroObject != null) {
-      ((PlatformData) stdData).setUnderlyingData(avroObject);
-      return stdData;
-    } else {
-      return null;
+  protected Object wrap(Object avroObject, Schema inputSchema, Object stdData) {
+    switch (inputSchema.getType()) {
+      case INT:
+      case LONG:
+      case BOOLEAN:
+        return avroObject;
+      case STRING:
+        return avroObject == null? null : avroObject.toString();
```

Comment:
Do null checks at the beginning so that its not repeated?

Reply:
It saves a conditional/branching in case those objects are primitives.